api/woeip/apps/air_quality/serializers.py

```diff
@@ -122,6 +130,21 @@ def create(self, validated_data):
             ends_at=validated_data.get("ends_at"),
         )
 
+        # Create file hash, check if hash already exists.
+        dusktrak_upload_hash = str(hash(dustrak_upload_file.read()))
+        hashes = set([filehash.hash for filehash in FileHash.objects.iterator()])
+        if dusktrak_upload_hash in hashes: #If hash exists, stop file upload.
```

Comment:
```suggestion
        if FileHash.objects.filter(hash=dustrak_upload_hash).exists(): #If hash exists, stop file upload.
```

Reply:
This is perfect! I didn't realize we could directly filter on the FileHash object without calling all(); hence why I resorted to the conversion of a set.